### PR TITLE
feat: do not render fees for de(registration) activities

### PR DESCRIPTION
**Pull request by xdzurman**
# Checklist

- [x] [JIRA](https://input-output.atlassian.net/browse/LW-8253?atlOrigin=eyJpIjoiNThmZmVjYThjNGEzNDc2ZDk4OWYyMDAzNGI5YjYzZDQiLCJwIjoiaiJ9)
- [ ] Proper tests implemented
- [x] Screenshots added.

---

## Proposed solution

Remove fees from (de)registartion activities. Edge case: standalone pool de-registrations occurring when delegating from pools ABC to pools AB (no delegation certificates). 

## Testing

Acitivity screen - check `Stake Key Registration` & `Stake Key De-Registration` activities, make sure they do not show `Transaction fee` information. Edge case: standalone De-registration activities (ones that are not accompanied by a same-timestamp delegation activity) (e.g. delegations from pools ABC to AB) should show fees.

## Screenshots

No tx fee:
image
image
No tx fee:
image
image

Example of edge case standalone de-registration:
image
image

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/linux/chrome/2049/6219462027/index.html) for [da13be3c](https://github.com/input-output-hk/lace/pull/512/commits/da13be3c94ef40ce8923d0647a846441aa3c2c02)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 31     | 1      | 0       | 0     | 32    | ❌     |

### CODE REVIEW

File: packages/core/src/ui/components/Transactions/Transaction.tsx [hunk 1]
```diff
@@ -258,27 +258,30 @@ export const Transaction = ({
...
-            </Tooltip>
-          </div>
 
-          <div data-testid="tx-fee" className={styles.detail}>
-            <div className={styles.amount}>
-              <span data-testid="tx-fee-ada" className={styles.ada}>{`${fee} ${coinSymbol}`}</span>
-              <span data-testid="tx-fee-fiat" className={styles.fiat}>
-                {amountTransformer(fee)}
-              </span>
+        {fee && fee !== '-' && (
```
Comment (xdzurman) [inline packages/core/src/ui/components/Transactions/Transaction.tsx:262]: this file's only addition is this wrapper. Otherwise with no fees, it'd look something like this
image

File: apps/browser-extension-wallet/src/stores/slices/transaction-detail-slice.ts [hunk 2]
```diff
@@ -134,6 +134,19 @@ const getTransactionDetail =
       (certificate) => certificate.__typename === 'StakeDelegationCertificate'
     ) as Wallet.Cardano.StakeDelegationCertificate[];
 
+    if (
```
Comment (refi93) [inline apps/browser-extension-wallet/src/stores/slices/transaction-detail-slice.ts:137]: for the sake of readability I'd consider extracting this to a `shouldOmitFee()` helper declared within this file, which then could be called a few lines above:

```
  fee: shouldOmitFee(tx) ? feeInAda : undefined
```

which would imho make for a slightly easier to follow code without bloating this already big and relatively complex function further

**Review by refi93 — APPROVED** [review]
only a minor refactor suggestion, otherwise LGTM 👍

**Review by przemyslaw-wlodek — APPROVED** [review]
